airflow/include/tasks/extract/utils/weaviate/ask_astro_weaviate_hook.py

```diff
@@ -315,30 +324,48 @@ def handle_upsert_rollback(
 
         for uuid in rollback_objects:
             try:
-                if self.client.data_object.exists(uuid=uuid, class_name=class_name):
+                if self.client.data_object.exists(uuid=uuid, class_name=class_name, tenant=tenant):
                     self.logger.info(f"Removing id {uuid} for rollback.")
-                    self.client.data_object.delete(uuid=uuid, class_name=class_name, consistency_level="ALL")
+                    self.client.data_object.delete(
+                        uuid=uuid, class_name=class_name, tenant=tenant, consistency_level="ALL"
+                    )
                 elif verbose:
                     self.logger.info(f"UUID {uuid} does not exist. Skipping deletion during rollback.")
             except Exception as e:
                 rollback_errors.append({"uuid": uuid, "result": {"errors": str(e)}})
                 if verbose:
                     self.logger.info(f"Error in rolling back id {uuid}. Error: {str(e)}")
 
-        for uuid in delete_objects:
+        return rollback_errors, delete_objects
+
+    def handle_successful_upsert(
+        self, objects_to_remove: list, class_name: str, verbose: bool, tenant: str | None = None
+    ) -> list:
+        """
+        Handles removal of previous objects after successful upsert.
+
+        :param class_name: Name of the class in Weaviate.
```

Comment:
I would suggest we make the docstring order consistent with the arguments

Reply:
Done